### PR TITLE
Clarify Node version support policy for package, and test in all supported versions

### PR DESCRIPTION
**Pull request by paulmelnikow**
Based on #3404, the four versions we support for the npm package are 8, 10, 11, and 12. Currently Node 8 is tested in our **main** build and Node 12 in **main@node-latest**. I added package-specific builds for 10 and 11.

We'll have to keep an eye on this as we fiddle with other builds. It could be easier if we added separate builds for each one, but having so many builds seems a bit unwieldy. Alternatively we could make a single build for the package, and use nvm or something like it to switch versions within that build.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -109,36 +109,67 @@ services_steps: &services_steps
     - store_test_results:
         path: junit
 
-jobs:
-  npm-install:
```
Comment (paulmelnikow) [inline .circleci/config.yml]: I commented out the **npm-install** job, which we're not using right now.

**Comment by paulmelnikow** [comment]
> Happy to merge this as it is, but I think this in the long run it will be easier if the range of node versions we test against is explicitly defined as opposed to implicit.

Pretty much agreed with that.

> FWIW, I don't see an issue with having an individual build for each node version.

My concern is that our build latency goes up even higher. (I think CircleCI gives us four simultaneous containers, which is why we often see "waiting behing your other queued builds".) Also scrolling through a long list in github is a bit harder.

Maybe we should attempt the nvm version?

> if we can only run the package tests on all the node versions if the PR touches files in the `gh-badges` dir.

I'm not sure there's a way to do that in Circle without actually starting a container. Also the list of files might be hard to contain. There's the root package.json, package-lock.json, and anything else that might affect the test toolchain.

**Comment by calebcartwright** [comment]
> I think CircleCI gives us four simultaneous containers, which is why we often see "waiting behing your other queued builds"

I was wondering about this the other day too as it seems like the PR checks have been slower of late (although I know Circle had some platform issues recently too that may have been a factor). Do they offer any additional containers/benefits for OSS projects under any circumstances? 

Also, should we look into seeing if we can skip the cypress binary download (or ensure it is cached with the other contents), outside the job that's running the e2e tests? It seems like the npm install step is often the longest step in those jobs (45-60 seconds), and I suspect that may be a factor

**Comment by chris48s** [comment]
> I'm not sure there's a way to do that in Circle without actually starting a container

I was thinking the same sort of thing as we do for the service tests, but you're right that initializing the container to decide whether to run the tests probably takes longer than running the tests in this case :D

**Comment by paulmelnikow** [comment]
Superseded by #3498.

**Pull request closed without merging**